### ISSUE TITLE
6.13.1 Parallel=Methods: test methods are executed one by one

### ISSUE THREAD

**Issue by rayzhouzhj** [cbeust/testng#1672]
### TestNG Version
6.13.1

### Expected behavior
Parallel execution of each method

### Actual behavior
1. Test methods executed in sequence
2. Parallel=Methods works fine in version 6.11
### Is the issue reproductible on runner?

- [ ] Shell
- [x] Maven
- [ ] Gradle
- [ ] Ant
- [ ] Eclipse
- [ ] IntelliJ
- [ ] NetBeans

### Test case sample
```
package com.practice.parallel;

import org.testng.annotations.Test;

public class TestNGTtest 
{
	@Test
	public void test1() throws InterruptedException
	{
		System.out.println("Test1 started");
		Thread.sleep(3000);
		System.out.println("Test1 completed");
	}
	
	@Test
	public void test2() throws InterruptedException
	{
		System.out.println("Test2 started");
		Thread.sleep(3000);
		System.out.println("Test2 completed");
	}
}
```
```
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
<suite name="Suite1" verbose="1" parallel="methods" thread-count="5">
  <test name="Regression"   >
    <packages>
      <package name="com.practice.parallel" />
   </packages>
 </test>
</suite>
```

**Comment by krmahadevan** [comment]
Duplicate of #1636

This has been fixed in `6.14-SNAPSHOT` and will be available in `6.14` release.

In the meantime, please use [this](https://github.com/cbeust/testng/issues/1636#issuecomment-349182219) simple workaround to get past the problem.

**Comment by rayzhouzhj** [comment]
Thanks @krmahadevan
